compliance_checker/imos/imos.py

```diff
@@ -369,31 +366,68 @@ def check_time_coverage(self, dataset):
                                 BaseCheck.HIGH)
 
         if result.value:
+            date_attribute_format = '%Y-%m-%dT%H:%M:%SZ'
+
+            time_var = dataset.dataset.variables.get('TIME', None)
+            time_min = np.amin(time_var.__array__())
+            time_max = np.amax(time_var.__array__())
+
+            time_units = getattr(time_var, "units", None)
+            time_calendar = getattr(time_var, "calendar", None)
+
             results = self._check_str_type(dataset, 'time_coverage_start')
             result = results[0]
             if result.value:
                 result_name = ('globalattr', 'time_coverage_start','check_date_format')
                 result = check_value(('time_coverage_start',),
-                                    '%Y-%m-%dT%H:%M:%SZ',
+                                    date_attribute_format,
                                     IMOSCheck.OPERATOR_DATE_FORMAT,
                                     dataset,
                                     IMOSCheck.CHECK_GLOBAL_ATTRIBUTE,
                                     result_name,
                                     BaseCheck.HIGH)
-            ret_val.append(result)
+            if result: 
+                ret_val.append(result)
+
+            if result.value:
+                time_coverage_start_string = getattr(dataset.dataset, "time_coverage_start", None)
+                time_coverage_start_datetime = datetime.strptime(time_coverage_start_string, date_attribute_format)
```

Comment:
Maybe redundancy here is not such a big deal. At least we don't have to maintain twice the same bit of code in util.py and imos.py.